src/Cli/func/Helpers/AzureHelper.cs

```diff
@@ -383,31 +383,49 @@ internal static async Task CheckFunctionHostStatusForFlex(Site functionApp, stri
             functionAppReadyClient.DefaultRequestHeaders.Add("User-Agent", Constants.CliUserAgent);
             functionAppReadyClient.DefaultRequestHeaders.Add("Accept", jsonContentType);
 
+            var statusUri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+            ColoredConsole.WriteLine($"Target endpoint: {statusUri.GetLeftPart(UriPartial.Path)}");
+
             await RetryHelper.Retry(
                 async () =>
                 {
+                    // Remove the header first if it already exists from a previous retry
+                    if (functionAppReadyClient.DefaultRequestHeaders.Contains("x-ms-request-id"))
+                    {
+                        functionAppReadyClient.DefaultRequestHeaders.Remove("x-ms-request-id");
+                    }
+
                     functionAppReadyClient.DefaultRequestHeaders.Add("x-ms-request-id", Guid.NewGuid().ToString());
-                    var uri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+
                     var request = new HttpRequestMessage()
                     {
-                        RequestUri = uri,
-                        Method = HttpMethod.Get
+                        RequestUri = statusUri,
+                        Method = HttpMethod.Get,
                     };
 
-                    var response = await functionAppReadyClient.SendAsync(request);
-                    ColoredConsole.Write(".");
+                    DateTime startTime = DateTime.UtcNow;
+
+                    HttpResponseMessage response = await functionAppReadyClient.SendAsync(request);
+
+                    TimeSpan elapsed = DateTime.UtcNow - startTime;
+                    ColoredConsole.WriteLine($"Response received after {elapsed.TotalMilliseconds:F0}ms - Status: {response.StatusCode} ({(int)response.StatusCode})");
 
                     if (response.IsSuccessStatusCode)
                     {
-                        ColoredConsole.WriteLine(" done");
+                        ColoredConsole.WriteLine("Host status check succeeded.");
```

Comment:
Include status code or if we get the host status json should we write that here?